test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -622,24 +622,25 @@ private static String groupName(ThreadGroup threadGroup) {
      * recursive shuffling behavior can be made by passing in the names of fields which
      * internally should stay untouched.
      */
-    public static XContentBuilder shuffleXContent(XContentBuilder builder, Set<String> exceptFieldNames) throws IOException {
+    public static XContentBuilder shuffleXContent(XContentBuilder builder, String... exceptFieldNames) throws IOException {
         BytesReference bytes = builder.bytes();
         XContentParser parser = XContentFactory.xContent(bytes).createParser(bytes);
         // use ordered maps for reproducibility
         Map<String, Object> shuffledMap = shuffleMap(parser.mapOrdered(), exceptFieldNames);
-        XContentBuilder jsonBuilder = XContentFactory.contentBuilder(builder.contentType());
+        XContentBuilder jsonBuilder = XContentFactory.contentBuilder(builder.contentType()).prettyPrint();
         return jsonBuilder.map(shuffledMap);
     }
 
-    private static Map<String, Object> shuffleMap(Map<String, Object> map, Set<String> exceptFieldNames) {
+    private static Map<String, Object> shuffleMap(Map<String, Object> map, String... exceptFieldNames) {
         List<String> keys = new ArrayList<>(map.keySet());
+        Set<String> exceptFields = exceptFieldNames == null ? Collections.emptySet() : new HashSet<>(Arrays.asList(exceptFieldNames));
```

Comment:
So maybe you don't need to handle the null case at all and just expect people not to pass null because it is a vararg. And it isn't passed as `@Nullable`.